/**
 * e8yes demo web server.
 *
 * <p>Copyright (C) 2020 Chifeng Wen {daviesx66@gmail.com}
 *
 * <p>This program is free software: you can redistribute it and/or modify it under the terms of the
 * GNU General Public License as published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * <p>You should have received a copy of the GNU General Public License along with this program. If
 * not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto3";

package e8;

import "file.proto";
import "user_profile.proto";

enum MessageChannelMemberType {
    MCMT_INVALID = 0;
    MCMT_MEMBER = 1;
    MCMT_ADMIN = 2;
}

message MessageChannelRelation {
    MessageChannelMemberType member_type = 1;
    int64 join_at = 2;
}

message MessageChannel {
    int64 channel_id = 1;
    string title = 2;
    string description = 3;
    FileTokenAccess avatar_readonly_access = 4;
    FileTokenAccess avatar_preview_readonly_access = 5;
    int64 created_at = 6;
}

message MessageChannelOveriew {
    MessageChannel channel = 1;
    int64 channel_last_interacted_at = 2;
    MessageChannelRelation channel_relation = 3;
    repeated UserPublicProfile most_active_users = 4;
}
